lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -4335,6 +4365,8 @@ class AdjointEmitter final : public SILInstructionVisitor<AdjointEmitter> {
   //    Adjoint: dealloc_stack adj[y]
   void visitAllocStackInst(AllocStackInst *asi) {
     auto adjBuf = getAdjointBuffer(asi);
+    if (errorOccurred)
```

Comment:
It's unfortunate that you have to check this in every visitor. I think a lot of changes in this patch will be refactored, but given the time constraint we can check them in.

Reply:
Yes, the checks everywhere are unfortunate - I hope we can remove them soon.